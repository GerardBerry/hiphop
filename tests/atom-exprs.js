"use strict"

var hh = require("hiphop");

var func = function(arg1, arg2) {
   console.log("atom works!", arg1, arg2);
}

var prg = <hh.module>
  <hh.loop>
    <hh.localsignal name="L" valued>
      <hh.emit signal="L" arg=${"foo bar"}/>
      <hh.pause/>
      <hh.atom func=${func} arg0=1 arg1=${hh.value("L")}/>
    </hh.localsignal>
  </hh.loop>
</hh.module>;

exports.prg = new hh.ReactiveMachine(prg, "atom");
