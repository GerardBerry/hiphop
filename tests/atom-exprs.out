atom> ;
--- Output:
atom works! 1 foo bar
atom> ;
--- Output:
atom works! 1 foo bar
atom> ;
--- Output:
atom works! 1 foo bar
atom> ;
--- Output:
atom works! 1 foo bar
atom> ;
--- Output:
atom works! 1 foo bar
atom> ;
--- Output:
atom works! 1 foo bar
atom> ;
--- Output:
atom works! 1 foo bar
atom> ;
--- Output:
atom works! 1 foo bar
atom> ;
--- Output:
atom works! 1 foo bar
atom> ;
--- Output:
